File: 111.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int res = INT_MAX;
    int minDepth(TreeNode* root) {
        helper(root, 1);
        return res;
    }
    void helper(TreeNode *node, int depth){
        if(node == NULL)
            return;
        if(!node->left && !node->right)
            res = min(res, depth);
        helper(node->left, depth+1);
        helper(node->right, depth+1);
    }
        
};

File: 111.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        st = [(root, 1)]
        res = float('inf')
        while st:
            cur, depth = st.pop()
            
            if not cur.left and not cur.right:
                res = min(res, depth)
            if cur.left:
                st.append((cur.left, depth + 1))
            if cur.right:
                st.append((cur.right, depth + 1))
        
        return res
            